torch/csrc/mps/Module.cpp

```diff
@@ -0,0 +1,85 @@
+#include <ATen/ATen.h>
+#include <ATen/mps/MPSAllocatorInterface.h>
+#include <ATen/mps/MPSGeneratorImpl.h>
+
+#include <torch/csrc/Generator.h>
+#include <torch/csrc/python_headers.h>
+#include <torch/csrc/utils/pybind.h>
+#include <torch/csrc/utils/python_numbers.h>
+#include <torch/csrc/utils/python_strings.h>
+
+using namespace torch;
+
+static PyObject* MPSModule_initExtension(PyObject* self, PyObject* noargs) {
+#if C10_ASAN_ENABLED
+  TORCH_WARN(
+      "torch.mps: your pytorch binary has address sanitizer (asan) built in, "
+      "asan is currently not compatible with torch.mps module, "
+      "you might get unexpected behavior (eg. out of memory, crash, etc.), "
+      "please rebuild pytorch without asan if you need to use this module");
+#endif
+  HANDLE_TH_ERRORS
+
+  auto m = THPObjectPtr(PyImport_ImportModule("torch.mps"));
+  if (!m)
+    throw python_error();
+
+  auto set_module_attr = [&](const char* name, PyObject* v) {
+    if (PyObject_SetAttrString(m, name, v) < 0) {
+      throw python_error();
+    }
+  };
+
+  // NOLINTNEXTLINE(performance-unnecessary-copy-initialization)
+  auto gen = at::mps::detail::getDefaultMPSGenerator();
+  auto default_mps_generator =
+      (THPGenerator*)THPGenerator_initDefaultGenerator(gen);
+  set_module_attr("default_mps_generator", (PyObject*)default_mps_generator);
+
```

Comment:
This feels like an anti-pattern: we should not litter public namespace with native objects, but rather make a function call (say `torch._C.get_mps_default_generator`) and call it from appropriate functions in python

Reply:
Updated the PR.